### DISCUSSION TITLE
Call for testing Freeplane 1.11.8 with new node style "Flower"

### DISCUSSION THREAD

**Discussion by dpolivaev** [freeplane/freeplane#1518] [Announcements]
Preview 1.11.8-09 is probably the last preview before the stable release.
I am so keen on the new way to draw the borders that I added a new user-defined node style "Flower" to all standard templates. Try it out.

**Comment by aaa1386** [comment]
Good changes have happened based on this link (https://github.com/freeplane/freeplane/discussions/1514). The interference of the node border with the background node was solved.

  But there is a problem, if it is solved, it will be much more complete.
  When we make the node border thicker, it makes the node smaller, this causes problems in some cases.

When we increase the width and length of a node, it expands outwards. The same thing should happen to the border node. When we make it thicker, it should expand outwards from the node, not to the inside of the node. make the node thin.
In the video below, you can see the thinning of the node by thickening the border of the node:



https://github.com/freeplane/freeplane/assets/38586259/a4102981-781e-4989-81e2-e7f7c35d3414


This sometimes causes problems:
You can see an example of it below:

https://github.com/freeplane/freeplane/assets/38586259/33e4349e-9262-47d3-bcdc-c7a432f4695f

**Reply by dpolivaev** [reply]
In order to simplify the alignment, max node width, and min node width settings include the borders.
Also please remember my request: whenever you upload a video or a screenshot I almost always need the map shown in the video or in the screenshot.

**Reply by aaa1386** [reply]
[Uploading pp.zip…]()

if possible:
  Two modes for the user to choose from: one is that when we make the border of the node thicker, the body of the node becomes thinner, and the other is that when the border of the node is made thicker, the main body does not become thinner and the border expands outwards.

**Reply by dpolivaev** [reply]
I changed the layout: now if you define min node width < max node width, the gap is used so that increasing the border width does not reduce the size of the inner part of the node. However, I am not sure if it is useful or rather too complicated.

I would appreciate it if you could check the new behavior and comment on it. I either keep the change or revert it, depending on your feedback.

Preview 1.11.8_10

**Reply by aaa1386** [reply]
I didn't have much time to check, but it seems like a positive step.
thanks.
![image](https://github.com/freeplane/freeplane/assets/38586259/668c6012-d3e9-46d2-86f3-3a3d9da2c9ff)

**Reply by aaa1386** [reply]
![image](https://github.com/freeplane/freeplane/assets/38586259/f08fbfe6-4511-4f4c-ae2d-5ff4c5d33a65)

**Reply by aaa1386** [reply]
(((if possible:
Two modes for the user to choose from: one is that when we make the border of the node thicker, the body of the node becomes thinner, and the other is that when the border of the node is made thicker, the main body does not become thinner and the border expands outwards.)))



  Three days later I will do a full test of the changes made

**Reply by aaa1386** [reply]
[LLLL.zip](https://github.com/freeplane/freeplane/files/13284828/LLLL.zip)

**Reply by aaa1386** [reply]
NEW SHAPES   
[WWWW.zip](https://github.com/freeplane/freeplane/files/13285177/WWWW.zip)

**Reply by aaa1386** [reply]
![image](https://github.com/freeplane/freeplane/assets/38586259/b31be53f-e4c2-46ec-a2c2-db39e5695f17)

**Reply by aaa1386** [reply]
![image](https://github.com/freeplane/freeplane/assets/38586259/50c5c7ba-b7af-4656-add5-df8868c71682)

**Reply by aaa1386** [reply]
[OO.zip](https://github.com/freeplane/freeplane/files/13289638/OO.zip)

Explanations are given in the mental maps below.(Select the nodes)
About Selected style
The following idea can be useful:
1- "Highlight All Matching Node" coloring mode is added for the selected node. By choosing the desired color
2- Next to all the options in the selected node style, there should be a checkbox: Use Setting Node -
For example, in the style editing environment of the selected mode, when setting the width, if we activate the "Use setting node" option for the width of that node, the width of each node that is selected will not change.


The cloud mode for the selected node style should only cloud the selected node, not sub nodes


These are not important things, but it is good if it is corrected.

**Reply by aaa1386** [reply]
@dpolivaev : I changed the layout: now if ...
Does not work if "uniform" box not selected
[BUG.zip](https://github.com/freeplane/freeplane/files/13292931/BUG.zip)

**Reply by aaa1386** [reply]
![image](https://github.com/freeplane/freeplane/assets/38586259/21f8625d-75a2-40de-9b88-0c98663eefce)

I explained some problems that happen due to this action in link #1521.
   If the maximum and minimum width of the node becomes independent from the thickness of the node border, it is true that the alignment with the upper and lower nodes sometimes gets messed up, but the user can reduce the minimum width of the node so that the alignment happens again.
Instead, this idea does not create other problems.

**Reply by dpolivaev** [reply]
I do not think I can support your ideas because I do not understand your use cases at all.

**Comment by aaa1386** [comment]
I think:
1- There is no need for a search box, the same box in the toolbar is sufficient. It is only necessary that when we press the control F, the editing mode is activated in the toolbar.
2- After the search reaches the end of the mental map and by clicking "Next Find" a message indicating that the search will be made from the beginning of the mental map. On the contrary, there is a need for a message.
  There are other cases that I will mention later.

**Reply by euu2021** [reply]
> There is no need for a search box, the same box in the toolbar is sufficient. It is only necessary that when we press the control F, the editing mode is activated in the toolbar.

I don't understand. Is there something different about the search box in this preview version?

**Reply by dpolivaev** [reply]
Yes, now "enter" does not close it but performs "search next", and you need to click "escape" to close it. The change was suggested by ajaberi1360.

**Reply by euu2021** [reply]
> There is no need for a search box, the same box in the toolbar is sufficient. It is only necessary that when we press the control F, the editing mode is activated in the toolbar.

The search box is relevant when using undocked map views, because, on undocked map views, there are no toolbars.

**Reply by aaa1386** [reply]
We have two modes, one is to display the "filter toolbar" or not.
This work can be done in two environments, map views and the main environment.
There is no need for another search box in the mind map when the "filter toolbar" is shown.
  If the user stops displaying the  "filter toolbar", a search box is needed in the middle of the mind map.
  If possible, when we press the control F, if the "filter toolbar" is active, the search editing mode in the "filter toolbar" will be activated
  If the "filter toolbar" display is stopped, the search box can be displayed by pressing Control F. This can be done either in the main environment or in the map views environment.

Search box position:
Now it is exactly in the middle of the mental map and sometimes it covers the nodes.
It is better when the user changes the position of the search box by dragging, the software considers the same place for subsequent searches and the search box is displayed in that position.

**Reply by euu2021** [reply]
I will tell you how I search nodes in a big map, and it may help your situation.

There is an excellent add-on, called [Jumper](https://github.com/lilive/Freeplane-Jumper), that I like to use to search nodes. Unfortunately, it's a bit unmaintained, and it's now getting bugs when used with the Jump In feature. 

So, I've been trying, and enjoying, the vanilla Find and Replace feature (`Edit->Find->Find and replace…`).  Although the name may make you think that it's only useful when the user wants to replace words, actually it's good at doing simple searches, too. And, it "remembers" the last spot that the Find and Replace window opened, so I make it open on my secondary monitor, which solves that situation that you mentioned:

> It is better when the user changes the position of the search box by dragging, the software considers the same place for subsequent searches and the search box is displayed in that position.

**Reply by aaa1386** [reply]
thank you
I will check, unfortunately I don't have time now

**Reply by aaa1386** [reply]
@dpolivaev
#1498 

I gave an idea in link #1498.
I think I explained some parts badly.
It was very nice that the search box would disappear from the middle of the mind map by pressing the "enter" button.✅✅✅

  After the changes , the search box does not disappear by pressing enter, and we have to press the escape key to make it disappear. This is not good in my opinion. It is a bit complicated. The search box also covers some nodes and results in the middle of the page.
I thought the following idea is possible: that when we type a word in the search box after pressing the "enter" button, the box will disappear and the first result will be shown, in addition to this, by pressing the enter button for the second time, the next result will be displayed. The search will be shown and the next results will be revealed by pressing the "Enter" button. Like searching in the toolbar, which shows the next searches every time you press enter.

Therefore, if this is not possible, return the program to its previous state, i.e., the state in which by pressing the first enter, both the search box disappears and the first result is shown. For the next results, we can use the "Find Next" shortcut button.
I'm sorry.

**Reply by aaa1386** [reply]
In other words, everything is fine except that the box is not hidden by pressing "enter"

**Reply by dpolivaev** [reply]
I am reverting the change because you are not happy with it and nobody else wanted to keep it.

**Comment by aaa1386** [comment]
When the search has no matching results, the message box shows this content.
This is very good, but when we search from the toolbar, this does not happen.
![image](https://github.com/freeplane/freeplane/assets/38586259/cc3b5e1f-4f37-44c1-a538-a2dca37be071)

**Reply by dpolivaev** [reply]
It means that you have a choice. I am not going to change it.

**Reply by aaa1386** [reply]
@dpolivaev
I think I explained something wrong.
What I mean is that whenever we search for a word and that word does not exist in the mind map. The following message should be shown in all cases, even when we search from the toolbar.
![image](https://github.com/freeplane/freeplane/assets/38586259/d6125f40-1313-47ea-b404-ad0ceb32c94c)


If the message does not appear, in order to find out that the search has not yielded results, we must do the following:
1- The word we want to be searched. We write in the search field on the toolbar. Suppose that word does not exist in the mind map.
2- We press the enter button. No message that shows that the search has not been successful will not appear.
3- In order to realize that the search has no result, we must study the selected node in the mental map and see if the word we searched for is among the expressions of that node or not?
  After doing these things, we realize that that word does not exist in the mental map.!!!!

**Reply by aaa1386** [reply]
There is a difference that you mentioned:
  Pay attention to the figure below.
![image](https://github.com/freeplane/freeplane/assets/38586259/3dd5bbf8-49ad-4278-8aaa-55d6ad78949f)
  After clicking on the word "Find", the search box appears in the mental map, but if the search is done through the toolbar, the search box does not appear. The existence of this difference is enough

**Comment by Olifair** [comment]
Hello, I just installed the 1.11.8 version and I don't seem to be able to find the "flower" node shape. In the drop down list for node shapes I only have Fork, Bubble, Oval, Rectangle, wide hex, narrow hex, as parent and combined.  Or am I not looking in the right place ?

**Reply by i-plasm** [reply]
Hi, you need to play with some node style parameters for the node to resemble a flower. Mainly `Line width` and `Boder line type`. As an example, set the `Line width` to 22 pt, and the `Border line type` to dashed. The other node parameters can be changed to resemble different shapes and colors of flowers. If you use any "Standard template", it already comes with a node style called "Flower".
![flower](https://github.com/freeplane/freeplane/assets/144830522/5da52948-1e7e-4059-84bd-65270c2e76dd)
